react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx

```diff
@@ -31,6 +37,23 @@ export function RuleBasedOutputsSelector({
     fetchNextPage
   } = useAllMappedEquipmentAssetMappings(models);
 
+  const contextualizedAssetNodes =
+    assetMappings?.pages
+      .flat()
+      .flatMap((item) => item.assets)
+      .map(convertAssetMetadataKeysToLowerCase) ?? [];
+
+  const expressions = ruleSet?.rulesWithOutputs
+    .map((ruleSet) => ruleSet.rule.expression)
+    .filter(isDefined);
+  const timeseriesExternalIdsFromRule = traverseExpressionToGetTimeseries(expressions) ?? [];
```

Comment:
This is triggered on every re-render, so this (maybe some of the lines above too?) should be put into `useMemo`

Reply:
https://github.com/cognitedata/reveal/pull/4414/commits/45046b273630ff5a219303859c636e899d25cfd8